app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/InsufficientConfidenceLevelController.scala

```diff
@@ -49,37 +58,90 @@ class InsufficientConfidenceLevelController @Inject()(
+      val form = hasNino.fold(haveANinoForm)(haveANinoForm.fill)
+      Ok(doYouHaveANinoPage(form))
     }
   }
 
   def doYouHaveNINOSubmit(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
-    withInsufficientConfidenceLevelUser {
+    withInsufficientConfidenceLevelUser { insufficientConfidenceLevel =>
       InsufficientConfidenceLevelController.haveANinoForm.bindFromRequest().fold(
         e => BadRequest(doYouHaveANinoPage(e)),
         hasNino =>
-          if (hasNino) {
-            redirectToIv
-          } else {
-            Ok(s"Got answer $hasNino")
+          updateSession(sessionStore, request)(
+            _.copy(subscriptionStatus = Some(insufficientConfidenceLevel.copy(hasNino = Some(hasNino))))
+          ).map{
+            case Left(e) =>
+            logger.warn("Could not update session after has NINO page submit",e )
+            errorHandler.errorResult()
+
+            case Right(_) =>
+              if (hasNino) {
+                redirectToIv
+              } else {
+                Redirect(routes.InsufficientConfidenceLevelController.doYouHaveAnSaUtr())
+              }
+
+          }
+     )
+    }
+  }
+
+  def doYouHaveAnSaUtr(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withInsufficientConfidenceLevelUser{ case IndividualInsufficientConfidenceLevel(hasNino, hasSaUtr) =>
+      hasNino.fold(
+        SeeOther(routes.InsufficientConfidenceLevelController.doYouHaveNINO().url)
+      ){ _ =>
+        val form = hasSaUtr.fold(hasSaUtrForm)(s => hasSaUtrForm.fill(s.value))
+        Ok(doYouHaveAnSaUtrPage(form, routes.InsufficientConfidenceLevelController.doYouHaveNINO()))
+      }
+    }
+  }
+
+  def doYouHaveSaUtrSubmit(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withInsufficientConfidenceLevelUser { insufficientConfidenceLevel =>
+    insufficientConfidenceLevel.hasNino.fold[Future[Result]](
+      SeeOther(routes.InsufficientConfidenceLevelController.doYouHaveNINO().url)
+    ) { _ =>
+      hasSaUtrForm.bindFromRequest().fold(
+        e => BadRequest(doYouHaveAnSaUtrPage(e, routes.InsufficientConfidenceLevelController.doYouHaveNINO())),
+        maybeSautr =>
+          updateSession(sessionStore, request)(
+            _.copy(subscriptionStatus = Some(insufficientConfidenceLevel.copy(hasSautr = Some(HasSAUTR(maybeSautr)))))
+          ).map {
+            case Left(e) =>
+              logger.warn("Could not update session after has SAUTR page submit", e)
+              errorHandler.errorResult()
+
+            case Right(_) =>
+              maybeSautr.fold(
+                Redirect(routes.RegistrationController.startRegistration())
+              ){ sautr =>
+              // TODO: get BPR here
+                Ok(s"Your Self Assessment Unique Taxpayer Reference is ${sautr.value}, call bpr")
```

Comment:
this needs to be implemented - I was going to do this after this chink of work. This changes in this branch were starting to get a bit big.

Reply:
Is there a ticket for work to call the ETMP with the SAUtr so we don't forget.

Reply:
it's part of the same ticket - I just thought it'd be easie/quickerr to review in separate chunks. I'm working on it right now